lib/streamlit/caching/cache_utils.py

```diff
@@ -228,6 +252,40 @@ def _show_cached_st_function_warning(
             dg.exception(e)
 
 
+class CachedFunctionMessagesCallStack(threading.local):
```

Comment:
This name isn't great, especially with CachedFunctionCallStack still hanging around. One or both should get a more informative name